tensorflow/core/kernels/eigen_spatial_convolutions-inl.h

```diff
@@ -21,6 +21,37 @@ namespace Eigen {
 
 namespace internal {
 
+// OptimizedPacketLoadOverTwoColumns<TensorEvaluatorType, PacketType, IndexType,
+// MaskType> provides `value` that is true if TensorEvaluatorType has
+// `PacketType partialPacket(IndexType, MaskType) const` and if the PacketType
+// supports masked load. In this case, we can split the packet over two columns
+// and use partial loads for each individual part before combining them to get
+// the required packet. This class is used to pick the correct implementation
+// of loadPacketStandard function.
+template <typename TensorEvaluatorType, typename PacketType, typename IndexType,
+          typename MaskType>
```

Comment:
I believe you are correct, it is passing SFINAE by conversion to `uint8_t` for AVX2 and `uint16_t` for AVX512 .
I agree that I should no longer need `tyepname MaskType` here since now the mask type is `typedef` in `unpacket_traits`. I missed updating this part when I made that particular change in the Eigen API. I will make the changes.